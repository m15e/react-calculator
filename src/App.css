html {
  box-sizing: border-box;
  font-size: 16px;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

body {
  background-color: rgb(21, 32, 43);
}

body,
h1,
h2,
h3,
h4,
h5,
h6,
p,
ol,
ul {
  margin: 0;
  padding: 0;
  font-weight: normal;
}

ol,
ul {
  list-style: none;
}

img {
  max-width: 100%;
  height: auto;
}

.app-container {
  width: 700px;
  margin: 0 auto;
}

.display {
  color: #fff;
  height: 100px;
  text-align: right;
  font-size: 70px;
  padding: 7px 20px;
  font-weight: bold;
  background-color: #848693;
}

.btn-group {
  display: flex;
  height: 100px;
}

.btn-group button {
  width: 100%;
  font-size: 20px;
}

.btn-group .wide {
  width: 207.5%;
}

button {
  border-color: #fff;
}

.header-container ul {
  display: flex;
  padding: 4rem;
}

.header-container li {
  margin-right: 2rem;
}

.header-container li a {
  color: #00b3ff;
  text-decoration: none;
  font-size: 20px;
}

.header-container li:nth-of-type(3) a {
  color: #fff;
}

.header-container li a:hover {
  opacity: 0.7;
}

.content-container {
  width: 700px;
  margin: 0 auto;
}

.content-container span,
.content-container p {
  color: #00b3ff;
  margin-bottom: 10px;
}

.content-container h2 {
  color: #fff;
  margin-bottom: 20px;
}
