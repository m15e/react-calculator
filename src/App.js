import '../App.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
// Line below is commented out as it is a project requirement that throws a linter error
// import calculate from '../logic/calculate';

function App() {
  return (
    <>
      <Display result="12" />
      <ButtonPanel />
    </>
  );
}

export default App;
