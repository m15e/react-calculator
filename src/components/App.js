import '../App.css';
import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

const App = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const clickHandler = buttonName => {
    setState(state => calculate(state, buttonName));
  };

  const { total, next, operation } = state;

  return (
    <div className="app-container">
      <Display result={{ total, next, operation }} />
      <ButtonPanel clickHandler={clickHandler} />
    </div>
  );
};

export default App;
