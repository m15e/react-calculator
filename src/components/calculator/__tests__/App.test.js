import React from 'react'
import renderer from 'react-test-renderer';
import App from '../App';

it('App component matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});