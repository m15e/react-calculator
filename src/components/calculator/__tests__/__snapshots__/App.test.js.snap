// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
Array [
  <div
    className="header-container"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Math Magicians
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="app-container"
  >
    <div
      className="display"
    >
      <p
        className="display-content"
      >
        0
      </p>
    </div>
    <div>
      <div
        className="btn-group"
      >
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          AC
        </button>
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          +/-
        </button>
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          %
        </button>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f39148",
            }
          }
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          7
        </button>
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          8
        </button>
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          9
        </button>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f39148",
            }
          }
          type="button"
        >
          X
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          4
        </button>
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          5
        </button>
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          6
        </button>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f39148",
            }
          }
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          1
        </button>
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          2
        </button>
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          3
        </button>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f39148",
            }
          }
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className="wide"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          0
        </button>
        <button
          className=""
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          .
        </button>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f39148",
            }
          }
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>,
]
`;
